### PR TITLE
Added implementation for download the amqstreams tarball from RH CSP

### PR DESCRIPTION
**Pull request by mronconis**
The implementation require to define "kafka_amqstreams_version" variable for install AMQ Streams or nothing for install kafka comunity.

## Issue
redhat_csp_download - Red Hat Accounts fail due to SSO change: https://github.com/ansible-middleware/redhat-csp-download/issues/20

### CODE REVIEW

File: roles/kafka_install/defaults/main.yaml [hunk 1]
```diff
@@ -12,9 +12,23 @@ kafka_group: kafka
 kafka_version: 3.1.1
 kafka_scala_version: 2.13
 
+kafka_rhn_id: 104335
```
Comment (saiello) [inline roles/kafka_install/defaults/main.yaml]: Users should not be aware of the RHN_ID. 

The "interface" might be simply:
```
kafka_version: <valid kafka version> 
amqstreams_enable: <boolean to toggle upstream/rh setup>
```

According to the provided variables and the "catalog" inside https://github.com/saiello/kafka/blob/main/roles/kafka_install/vars/main.yaml proper rhn_id and other stuff are resolved.

Comment (saiello) [reply roles/kafka_install/defaults/main.yaml]: see https://github.com/saiello/amq-streams

Comment (mronconis) [reply roles/kafka_install/defaults/main.yaml]: Simplified the interface and revisited the catalog structure as below:
```
kafka_versions:
  3.2.3:
    scala_version: 2.13
    upstream:
      version: 3.2.3 (Commit:50029d3ed8ba576f)
    amqstreams:
      version: 3.2.3.redhat-00003 (Commit:8c8f89d8e34196ab)
      product_version: 2.2.0
      rhn_id: 104745
```

Users will only need to configure:
```
kafka_version: <valid kafka version> (by default use latest version defined on the catalog)
amqstreams_enable: <boolean to toggle upstream/rh setup>
```

**Comment by mronconis** [comment]
The updated catalogue versions is as follows:
```
# upstream catalogue
kafka_versions:
  3.2.3:
    scala_version: 2.13
    microversion: 3.2.3 (Commit:50029d3ed8ba576f)
```
and
```
# amqstreams catalogue
amqstreams_versions:
  2.2.0:
    scala_version: 2.13
    microversion: 3.2.3.redhat-00003 (Commit:8c8f89d8e34196ab)
    rhn_id: 104745
```

Users will only need to configure:
```
# upstream setup
kafka_version: <valid kafka version> (by default use latest version defined on the catalog)
```
or
```
# amqstreams setup
amqstreams_enable: true
amqstreams_version: <valid amq streams version> (by default use latest version defined on the catalog)
```

**Pull request merged**